File: src/Bolt.Core/Entities/Menu.cs
using System;
using System.Collections.Generic;
using Bolt.SharedKernel;

namespace Bolt.Core.Entities
{
    public class Menu : BaseEntity
    {
        public Guid RestaurantId { get; set; }
        public Restaurant Restaurant { get; set; }
        public ICollection<MenuCategory> Categories { get; set; }
    }
}

File: src/Bolt.Web/wwwroot/src/models/Menu.ts
import { Type } from 'class-transformer';
import { type } from 'os';
import { BaseEntity } from './BaseEntity';
import { MenuCategory } from './MenuCategory';
import { Restaurant } from './Restaurant';

export class Menu extends BaseEntity {
  restaurant?: Restaurant;

  @Type(() => MenuCategory)
  categories: MenuCategory[] = [];
}
